facter/facter.go

```diff
@@ -197,6 +209,28 @@ func (f *Facter) GetConfigPolicy() (*cpolicy.ConfigPolicy, error) {
 
 // ------------ helper functions --------------
 
+// createMetricsSubArray creates an array of metrics for map values returned by facter
+func createMetricsSubArray(name string, value map[string]interface{}) ([]plugin.MetricType, error) {
+
+	var ms []plugin.MetricType
+
+	namespaces := []string{}
+
+	err := utils.FromMap(value, name, &namespaces)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, namespace := range namespaces {
+		ns := strings.Split(namespace, "/")
```

Comment:
validateNamespace expects namespace to contain exactly 3 segments, GetMetricTypes return such namespaces but here you are expecting additional elements of namespace. They cannot be collected if not reported by GetMetricTypes.